controllers/photoController.js

```diff
@@ -2,33 +2,131 @@ const url = require("url");
 
 const Photo = require("../models/Photo");
 const Point = require("../models/Point");
+const Map = require("../models/Map");
+const Comment = require("../models/Comment");
+
 const { ERROR_MESSAGE } = require("../constants");
 
+const uploadPhoto = async (req, res, next) => {
+  const { photo, point, description, map } = req.body;
+  const { createdAt, createdBy } = JSON.parse(photo);
+  const { latitude, longitude, placeName } = JSON.parse(point);
+
+  try {
+    const [currentPoint, currentMap] = await Promise.all([
+      Point.findOne({ latitude, longitude }).exec(),
+      Map.findById(map).exec(),
+    ]);
+    const newPhoto = new Photo({
+      createdAt,
+      createdBy,
+      url: req.file.location,
+      placeName,
+      description,
+    });
+
+    if (currentPoint) {
+      const currentPointId = currentPoint._id;
+      const newPhotoId = newPhoto._id;
+
+      currentPoint.photos.push(newPhotoId);
+      newPhoto.point = currentPointId;
+      currentMap.photos.push(newPhotoId);
+
+      await Promise.all([
+        newPhoto.save(),
+        currentPoint.save(),
+        currentMap.save(),
+      ]);
+
+      res.json({
+        result: "ok",
+      });
+    }
+
+    if (!currentPoint) {
+      const newPoint = await Point.create(point);
+      const newPointId = newPoint._id;
+      const newPhotoId = newPhoto._id;
+
+      currentMap.points.push(newPointId);
+      currentMap.photos.push(newPhotoId);
+      newPoint.photos.push(newPhotoId);
+      newPhoto.point = newPointId;
+
+      await Promise.all([newPhoto.save(), newPoint.save(), currentMap.save()]);
+
+      res.json({
+        pointId: newPointId,
+      });
+    }
+  } catch {
+    res.json({
+      error: {
+        message: ERROR_MESSAGE.SERVER_ERROR,
+        code: 500,
+      },
+    });
+  }
+};
+
 const deletePhoto = async (req, res, next) => {
   const { id: photoId } = req.params;
-  const { point: pointId } = url.parse(req.url, true).query;
+  const { map: mapId } = url.parse(req.url, true).query;
 
   try {
-    const point = await Point.findById(pointId).exec();
+    const [currentMap, currentPhoto] = await Promise.all([
+      Map.findById(mapId).exec(),
+      Photo.findById(photoId).populate("comments").exec(),
```

Comment:
```suggestion
      Photo.findById(photoId).populate("comments").populate("point").exec(),
```

이렇게 하면 85번째 줄에서 Point collection에서 pointInPhoto를 기반으로 한 번 더 안 찾아도 될 것 같은데
이렇게 하는건 어떨까요-?